tardis/visualization/tools/sdec_plot.py

```diff
@@ -424,6 +439,95 @@ def from_hdf(cls, hdf_fpath):
+                        element = species.split(" ")[0]
+                        # Next thing is the ion range
+                        # convert the requested ions into numerals
+                        first_ion_numeral = roman_to_int(
+                            species.split(" ")[-1].split("-")[0]
+                        )
+                        second_ion_numeral = roman_to_int(
+                            species.split(" ")[-1].split("-")[-1]
+                        )
+                        # add each ion between the two requested into the species list
+                        for ii in np.arange(
+                            first_ion_numeral, second_ion_numeral + 1
+                        ):
+                            full_species_list.append(
+                                element + " " + int_to_roman(ii)
+                            )
+                    else:
+                        # Otherwise it's either an element or ion so just add to the list
+                        full_species_list.append(species)
+
+                # full_species_list is now a list containing each individual species requested
+                # e.g. it parses species_list = [Si I - V] into species_list = [Si I, Si II, Si III, Si IV, Si V]
+                self._full_species_list = full_species_list
+                requested_species_ids = []
+                keep_colour = []
+
+                # go through each of the requested species. Check whether it is
+                # an element or ion (ions have spaces). If it is an element,
+                # add all possible ions to the ions list. Otherwise just add
+                # the requested ion
+                for species in full_species_list:
+                    if " " in species:
+                        requested_species_ids.append(
+                            [
+                                species_string_to_tuple(species)[0] * 100
+                                + species_string_to_tuple(species)[1]
+                            ]
+                        )
+                    else:
+                        atomic_number = element_symbol2atomic_number(species)
+                        requested_species_ids.append(
+                            [
+                                atomic_number * 100 + ii
+                                for ii in np.arange(atomic_number)
+                            ]
+                        )
+                        # add the atomic number to a list so you know that this element should
+                        # have all species in the same colour, i.e. it was requested like
+                        # species_list = [Si]
+                        keep_colour.append(atomic_number)
+                requested_species_ids = [
+                    species_id
+                    for list in requested_species_ids
+                    for species_id in list
+                ]
```

Comment:
It's not a reserved keyword but a built-in type (different things in Python). Not that I'm saying it's okay to redefine it here! Linters should catch it, maybe adding a note to TARDIS Developer Workflow guide about using a linter might be a good idea (would go right next to "run tests before making a PR"). Maybe even add pre-commit hook to run linter/Black/whatever.

Reply:
> It's not a reserved keyword but a built-in type (different things in Python).
@Molkree right - thanks for correcting.

> Not that I'm saying it's okay to redefine it here! Linters should catch it, maybe adding a note to TARDIS Developer Workflow guide about using a linter might be a good idea (would go right next to "run tests before making a PR")
Yes that's a good idea - I myself use pylint, @MarkMageeAstro you can try using that next time - VSCode already has option for it.

> Maybe even add pre-commit hook to run linter/Black/whatever.
Black is just a code formatter and we already have pipeline for it, we can use a pipeline for a linter too but it will require certain amount of work in configuring (keep it not too strict) and writing a gh-action. If you've time, feel free to submit a PR @Molkree